### DISCUSSION TITLE
[General] Anonymous Access Consideration

### DISCUSSION THREAD

**Discussion by aldbr** [DIRACGrid/diracx-web#17] [User personas and stories]
### User Persona

- Name: Edward Sunden
- Age: 40
- Occupation: Potential User/Visitor
- Goals:
  1. View some information without having to log in.
  2. Access monitoring data for quick reference or casual browsing.
- Pain points:
  1. Current system assumes users always have a token, making it impossible to view information without logging in.
  2. Lack of anonymous access might deter potential users or make quick referencing cumbersome.

### User Story

As a casual browser or potential user, I want the ability to view some information, like monitoring data, without having to log in, so that I can quickly reference necessary data or explore the application without commitment.

**Comment by aldbr** [comment]
- Given that monitoring data will be available elsewhere (Grafana), anonymous access does not seem to be necessary for the new application.

- If anonymous access is deemed necessary, changes in DiracX to make API routes public will be required, allowing users to access certain information without a token.

Before making any design changes, it's crucial to gather use cases and understand the actual need for anonymous access in the new application. This understanding will guide whether to implement changes in DiracX to support anonymous access or drop the feature altogether.

**Comment by chaen** [comment]
Conclusion: anonymous access is not needed. For site admins or others really wanting to dig into the system, they should be granted access via IAM/other.
